java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -1187,6 +1293,52 @@ public final Builder append(double value) {
       return this;
     }
 
+    public final Builder append(BigDecimal value) {
+      return append(value, null);
+    }
+
+    //
+    public final Builder append(BigDecimal value, RoundingMode roundingMode) {
+      assert type.isDecimalType();
+      assert currentIndex < rows;
+      BigInteger unscaledValue;
+      // rescaled input decimal if RoundingMode set
+      if (roundingMode == null) {
+        assert -type.getScale() == value.scale();
+        unscaledValue = value.unscaledValue();
+      } else {
+        unscaledValue = value.setScale(-type.getScale(), roundingMode).unscaledValue();
+      }
+      if (type.typeId == DType.DTypeEnum.DECIMAL32) {
+        data.setInt(currentIndex * type.getSizeInBytes(), unscaledValue.intValueExact());
+      } else if (type.typeId == DType.DTypeEnum.DECIMAL64) {
+        data.setLong(currentIndex * type.getSizeInBytes(), unscaledValue.longValueExact());
+      } else {
+        throw new IllegalStateException(type + " is not a supported decimal type.");
+      }
+      currentIndex++;
+      return this;
+    }
+
+    public final Builder appendUnscaledDecimal(int value) {
+      assert type.isDecimalType();
+      assert currentIndex < rows;
+      if (type.typeId == DType.DTypeEnum.DECIMAL64) {
```

Comment:
This should be the first check in the method, as `appendunscaledDecimal` will also check for the above assertions.

Reply:
Fixed. Thanks for advice!